backend/small_eod/dictionary/models.py

```diff
@@ -0,0 +1,26 @@
+from django.db import models
+from generic.models import TimestampUserLogModel
+from case.models import Case
+
+
+class DictTypes(models.TextChoices):
+    WHOSE = "WHOSE", "Do kogo należy sprawa"
+    WSCOPE = "WSCOPE", "Jaki zakres sprawy"
+    ISCOPE = "ISCOPE", "Bezczynność w jakim zakresie"
+    DSCOPE = "DSCOPE", "Decyzja w jakim zakresie"
+    INFOT = "INFOT", "Informacja udzielona w którym momencie"
+    STATUS = "STATUS", "Status"
+
+
+class Dictionary(TimestampUserLogModel):
+    type = models.CharField(max_length=6, choices=DictTypes.choices)
+    name = models.CharField(max_length=100)
+    active = models.BooleanField(default=False)
+    case = models.ForeignKey(to=Case, on_delete=models.CASCADE)
+
+
+class ChoiceRange(models.Model):
```

Comment:
Czy słowo "Range" jest w tym miejscu odpowiednie? Sam nie mam wiedzy o języku angielskim dostatecznej, ale range dla enum budzi moją wątpliwość. Poddaje pod wątpliwość, gdyż może ktoś inny rozstrzygnie, a jak nie to zostanie takie.

Reply:
Tutaj chcialbym sie na chwilke zatrzymac, zeby wytlumaczyc w jaki sposob to jest zaimplementowane wszystko (caly matrix).
Podzielilem go na 2 tabele, polaczone z Case za pomoca FK. 
1. Dictionary - ten model bedzie zbieral input uzytkownika - dla kazdego wyboru bedzie osobny dictionary
2.ChoiceRange - ten model bedzie dawal informacje o tym ile minimalnie i ile maksymalnie checkboxow uzytkownik moze zaznaczyc dla kazdego typu zdefiniowanego w DictTypes

Czyli:


```
case :{
	id:1
	createdBy:1
	...
	matrix:{
		dictionaries:[
			{type:WHOSE, name:'firma prywatna', active:True},
			{type:WHOSE, name:'firma nieprywatna', active:False},
			...
			{type:ISCOPE, name:'brak odpowiedzi', active:True},
			{type:ISCOPE, name:'iformacja nie stanowi IP', active:True},
			..
			{type:DSCOPE, name:'prywatnosc', active:True}
			],
		choice_ranges:[
			{type:WHOSE, minItems:1, maxItems:3},
			{type:ISCOPE, minItems:1, maxItems:1},
			{type:DSCOPE, minItems:1, maxItems:1}
			...
			]
	}
}
```

nie chcialem umieszczac min i max items w dictionary, poniewaz wydaje mi sie, ze to moze prowadzic do problemow z integralnoscia danych, jezeli w kazdym dictionary moznaby to bylo ustawic osobno (np dla 2 opcji z WHOSE mogloby sie zdazyc inne max i min items).

Prosze o komentarz

Reply:
@ad-m - to jest schowane, wiec taguje dla widocznosci

Reply:
Dziękuje, odniosę się do tego następnego wieczoru.